frontend/apps/pwa/src/components/formFields/WeekFieldEditor.tsx

```diff
@@ -1,38 +1,40 @@
 import React, { FC } from 'react';
 
 import { FieldEditorProps } from './types';
-import { FieldDescription } from './FieldDescription';
-import { FieldLabel } from './FieldLabel';
 
 export const WeekFieldEditor: FC<FieldEditorProps> = ({
   field,
   value,
   onChange,
+  register,
+  errors,
 }) => {
-  const isValidWeek = (weekString: string) => {
-    const weekRegex = /^(?:19|20|21)\d{2}-W[0-5]\d$/;
-    return weekRegex.test(weekString) && +weekString.slice(6) <= 52;
-  };
-
-  const onChangeHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
-    if (!isValidWeek(event.target.value)) return;
-    onChange(event.target.value);
-  };
-
+  const expectedLength = 8;
+  const registerObject = register(`values.${field.id}`, {
+    required: { value: field.required, message: 'This field is required' },
+    pattern: {
+      value: /^(?:19|20|21)\d{2}-W[0-5]\d$/,
+      message: 'wrong pattern',
+    },
+    maxLength: { value: expectedLength, message: 'Value is too long' },
```

Comment:
Do we need this if we have the pattern match? I think the message is misleading as the pattern is the priority.